pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -100,6 +100,20 @@ func (p *Predictor) Reconcile(isvc *v1beta1.InferenceService) error {
 				return fmt.Errorf("specified runtime %s does not support specified framework/version", *isvc.Spec.Predictor.Model.Runtime)
 			}
 
+			// TODO: adding mlserver specific environment variables, will be removed after mlserver issue fix
+			if *isvc.Spec.Predictor.Model.Runtime == "kserve-mlserver" && isvc.Spec.Predictor.Model.GetStorageUri() != nil {
+				isvc.Spec.Predictor.Model.Env = append(isvc.Spec.Predictor.Model.Env,
+					v1.EnvVar{
+						Name:  constants.MLServerModelNameEnv,
+						Value: isvc.Name,
+					},
```

Comment:
I think the `MLServerModelNameEnv` should be still in the mutator and only add if user does not specify the env variable, so later we can simply remove this block